tox.ini

```diff
@@ -48,5 +48,22 @@ deps =
     pep517
     twine
 commands =
+    python -c "from pathlib import Path; \
+               [x.unlink(missing_ok=True) for x in Path('{toxinidir}/dist').glob('*')]"
     python -m pep517.build -s -b {toxinidir} -o {toxinidir}/dist
     twine check {toxinidir}/dist/*
+
+[testenv:release]
+description = Make a release; must be called after "build"
+skip_install = True
+deps =
+    twine
+depends =
+    build
+setenv =
+    TWINE_USERNAME={env:TWINE_USERNAME:INVALID USERNAME}
+    TWINE_PASSWORD={env:TWINE_PASSWORD:INVALID PASSWORD}
+commands =
+    twine check {toxinidir}/dist/*
+    twine upload {toxinidir}/dist/* \
+                 {posargs:--repository-url=https://test.pypi.org/legacy/}
```

Comment:
Why don't you set this via an env var too?

Reply:
It defaults to uploading to the main PyPI, and I want to invert that default, even if running it from the command line.

Also I couldn't get it to work reliably the other way and if it fails it uploads to real PyPI 😬.

Reply:
It's not very explicitly documented but: https://github.com/pypa/gh-action-pypi-publish/blob/37e305e/twine-upload.sh#L34

Reply:
It's in the man page, but if you fail to set it correctly (which happened a few times in my tests), it uploads directly to pypi. This way fails safe.

Reply:
Ouch, that is unfortunate